pennylane/_device.py

```diff
@@ -744,19 +745,14 @@ def batch_transform(self, circuit: QuantumScript):
 
         hamiltonian_unusable = not supports_hamiltonian or (finite_shots and not is_shadow)
 
-        if hamiltonian_in_obs and (hamiltonian_unusable or (use_grouping and grouping_known)):
+        if (hamiltonian_in_obs and (hamiltonian_unusable or (use_grouping and grouping_known))) or (
+            expval_sum_in_obs and not is_shadow
+        ):
             # If the observable contains a Hamiltonian and the device does not
             # support Hamiltonians, or if the simulation uses finite shots, or
             # if the Hamiltonian explicitly specifies an observable grouping,
             # split tape into multiple tapes of diagonalizable known observables.
-            try:
-                circuits, hamiltonian_fn = qml.transforms.hamiltonian_expand(circuit, group=False)
-            except ValueError as e:
-                raise ValueError(
```

Comment:
Can we now return expectations values of multiple Hamiltonians? :)

Reply:
Yes!

Reply:
The following code raises an error in `master`
```python
import pennylane as qml

dev = qml.device("default.qubit", wires=2)
H = qml.PauliX(0) @ qml.PauliZ(1) + qml.PauliZ(1)
H.compute_grouping()

@qml.qnode(dev)
def circuit():
    return qml.expval(H), qml.expval(H)

print(circuit())
```

Reply:
That's a very good feature unlocked :100: